### PR TITLE
HPCC-22848 Build with Ubuntu 19.10/GNU GCC 9

### PR DESCRIPTION
**Pull request by richardkchapman**
Signed-off-by: Richard Chapman <rchapman@hpccsystems.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [x] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

## Testing:
Compiled using Docker image.

### CODE REVIEW

**Comment by richardkchapman** [comment]
@ghalliday For review

**Comment by richardkchapman** [comment]
@Michael-Gardner We may need a new dependencies file for Ubuntu 1910?

File: deployment/deployutils/CMakeLists.txt [hunk 1]
```diff
@@ -48,6 +48,10 @@ include_directories (
          ${HPCC_SOURCE_DIR}/system/security/shared
     )
 
+if (CMAKE_COMPILER_IS_GNUCC AND NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS "9.2.0")
+  set_source_files_properties(deployutils.cpp PROPERTIES COMPILE_FLAGS -Wno-format-overflow)
```
Comment (richardkchapman) [inline deployment/deployutils/CMakeLists.txt]: There seems to be a false positive error report in this file - I could not work out what the compiler was complaining about. GCC docs do mention that this warning can occasionally trigger false positives.

File: plugins/cassandra/CMakeLists.txt [hunk 2]
```diff
@@ -66,9 +66,9 @@ if(USE_CASSANDRA)
   if (CMAKE_COMPILER_IS_GNUCXX)
   # From time to time need to add flags in here to disable warnings, if a new compiler picks up something new.
   # e.g.
-  # if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 7.0.0)
-  #   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wimplicit-fallthrough=0)
-  # endif ()
+    if (NOT CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.2.0)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-deprecated-copy")
```
Comment (richardkchapman) [inline plugins/cassandra/CMakeLists.txt]: Should consider updating cassandra client libraries, but not at this late stage

File: system/jlib/jutil.cpp [hunk 3]
```diff
@@ -1680,11 +1680,13 @@ unsigned __int64 greatestCommonDivisor(unsigned __int64 left, unsigned __int64 r
 #endif
 
 //In a separate module to stop optimizer removing the surrounding catch.
+int minus4096 = -4096;  // Stops compiler being clever enough to report error
```
Comment (richardkchapman) [inline system/jlib/jutil.cpp]: Note - can't be static or compiler still notices the error

**Comment by xwang2713** [comment]
Test build run successfully with GNU GCC 9

**Comment by richardkchapman** [comment]
@ghalliday

**Pull request merged**